### PR TITLE
RPP Audio Support HIP - Mel Filter Bank

### PR DESCRIPTION
**Pull request by r-abishek**
- Updates version to RPP 1.9.2
- Adds RPP audio support for Mel Filter Bank on HIP backend.
- Adds relevant qa tests and performance tests.

### CODE REVIEW

File: utilities/test_suite/HIP/runAudioTests.py [hunk 1]
```diff
@@ -224,7 +223,7 @@ def rpp_test_suite_parser_and_validator():
 subprocess.call(["make", "-j16"], cwd=".")    # nosec
 
 # List of cases supported
-supportedCaseList = ['0', '1', '3']
+supportedCaseList = ['0', '1', '3', '7']
```
Comment (LakshmiKumar23) [inline utilities/test_suite/HIP/runAudioTests.py:226]: What about these updates for the HOST backend python script?
https://github.com/ROCm/rpp/blob/develop/utilities/test_suite/HOST/runAudioTests.py

Comment (r-abishek) [reply utilities/test_suite/HIP/runAudioTests.py:226]: HOST is already in ToT, this PR is only for HIP support.
HOST L179 -> https://github.com/ROCm/rpp/blob/develop/utilities/test_suite/HOST/runAudioTests.py#L179

File: utilities/test_suite/HOST/Tensor_audio_host.cpp [hunk 2]
```diff
@@ -361,45 +368,16 @@ int main(int argc, char **argv)
...
-                    unsigned long long spectrogramBufferSize = (unsigned long long)srcDescPtr->h * (unsigned long long)srcDescPtr->w * (unsigned long long)srcDescPtr->c * (unsigned long long)srcDescPtr->n;
-                    unsigned long long melFilterBufferSize = (unsigned long long)dstDescPtr->h * (unsigned long long)dstDescPtr->w * (unsigned long long)dstDescPtr->c * (unsigned long long)dstDescPtr->n;
-                    inputf32 = (Rpp32f *)realloc(inputf32, spectrogramBufferSize * sizeof(Rpp32f));
-                    outputf32 = (Rpp32f *)realloc(outputf32, melFilterBufferSize * sizeof(Rpp32f));
-
-                    // Read source data
-                    read_from_bin_file(inputf32, srcDescPtr, srcDimsTensor, "spectrogram", scriptPath);
+                    Rpp32s srcDimsTensor[batchSize * 2];
+                    // (height, width) for each tensor in a batch for given QA inputs.
+                    srcDimsTensor[0] = 257;
```
Comment (rrawther) [inline utilities/test_suite/HOST/Tensor_audio_host.cpp:373]: why is this changed to single assignments

File: utilities/test_suite/HOST/Tensor_audio_host.cpp [hunk 3]
```diff
@@ -361,45 +368,16 @@ int main(int argc, char **argv)
...
-
-                    // Set buffer sizes for src/dst
-                    unsigned long long spectrogramBufferSize = (unsigned long long)srcDescPtr->h * (unsigned long long)srcDescPtr->w * (unsigned long long)srcDescPtr->c * (unsigned long long)srcDescPtr->n;
-                    unsigned long long melFilterBufferSize = (unsigned long long)dstDescPtr->h * (unsigned long long)dstDescPtr->w * (unsigned long long)dstDescPtr->c * (unsigned long long)dstDescPtr->n;
-                    inputf32 = (Rpp32f *)realloc(inputf32, spectrogramBufferSize * sizeof(Rpp32f));
-                    outputf32 = (Rpp32f *)realloc(outputf32, melFilterBufferSize * sizeof(Rpp32f));
-
-                    // Read source data
-                    read_from_bin_file(inputf32, srcDescPtr, srcDimsTensor, "spectrogram", scriptPath);
+                    Rpp32s srcDimsTensor[batchSize * 2];
```
Comment (rrawther) [inline utilities/test_suite/HOST/Tensor_audio_host.cpp:371]: I think batch_size has to be 3 here. In that case it is better to use srcDimsTensor[6] = {..}; assignment here

**Review by rrawther — CHANGES_REQUESTED** [review]
please address the review comments and resolve conflicts if any

File: utilities/test_suite/HOST/Tensor_audio_host.cpp (hunk 3 above)
Comment (r-abishek) [reply utilities/test_suite/HOST/Tensor_audio_host.cpp]: resolved

File: utilities/test_suite/HOST/Tensor_audio_host.cpp (hunk 2 above)
Comment (r-abishek) [reply utilities/test_suite/HOST/Tensor_audio_host.cpp]: changed on host, doing it on hip may involve an additional hipmemcpy

File: CHANGELOG.md [hunk 4]
```diff
@@ -2,6 +2,12 @@
 
 Full documentation for RPP is available at [https://rocm.docs.amd.com/projects/rpp/en/latest](https://rocm.docs.amd.com/projects/rpp/en/latest)
  
+## RPP 1.9.2 (Unreleased)
+
+### Changes
```
Comment (r-abishek) [inline CHANGELOG.md:7]: @kiritigowda @rrawther @LakshmiKumar23 For every new functionality addition PR issued, we are now incrementing the version like this - 1.9.1 -> 1.9.2 (for melfilter bank) -> 1.9.3 (for spectrogram) -> 1.9.4 (for box filter).

Please see in the 3 open PRs - does this work?
The versioning is uniformly changed in 3 RPP files in every PR:
- CHANGELOG.md
- CMakeLists.txt
- include/rpp_version.h

**Comment by r-abishek** [comment]
@AryanSalmanpour This PR addresses verbal comments from the code review now

**Review by AryanSalmanpour — APPROVED** [review]

**Review by rrawther — APPROVED** [review]

**Comment by rrawther** [comment]
@kiritigowda : CI is still failing for bunch of RPP PRs and hence we can't merge. I think this is failing on RHEL and SLES. Can you please check with @r-abishek and see how to resolve this?

**Comment by rrawther** [comment]
@randyh62 : Can you please approve this. We need to merge this PR

**Comment by r-abishek** [comment]
All Jenkins CI passing. Waiting on Azure https://dev.azure.com/ROCm-CI/ROCm-CI/_build/results?buildId=10402&view=results

File: CHANGELOG.md (hunk 4 above)
Comment (LakshmiKumar23) [reply CHANGELOG.md:7]: @r-abishek RPP_VERSION_PATCH should be good. We can increment MINOR for a whole new functionality like Audio

**Review by LakshmiKumar23 — APPROVED** [review]

**Review by randyh62 — APPROVED** [review]
Looks good to me.
